django/db/migrations/loader.py

```diff
@@ -315,6 +319,32 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # Since we added "key" to the nodes before this implies
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
+        # Check that no migrations are applied before their dependencies
```

Comment:
Can you move the entire block into its own method on the `MigrationLoader`, e.g. `def check_consistent_history(self)` and call that from within `build_graph()`, please.